packages/nimble-components/src/text-field/styles.ts

```diff
@@ -104,6 +105,7 @@ export const styles = css`
         color: transparent;
         width: 0px;
         ${userSelectNone}
+        margin-left: -1px;
```

Comment:
Note: If we decide not to fix #2400, then this negative left margin can be removed.

Reply:
How would you feel about splitting out any border spacing stuff from this PR to avoid blocking on that discussion? i.e. hold the current assumption that spacing from the border does not include the border for now.

Reply:
That seems like a reasonable plan to me.